tispark-overview.md

```diff
@@ -32,308 +32,407 @@ Also, TiSpark supports distributed writes to TiKV. Compared with writes to TiDB
 
-val customer = spark.sql("select * from customer limit 100000")
-// You might repartition the source to make it balance across nodes
-// and increase the concurrency.
-val df = customer.repartition(32)
-df.write
-.mode(saveMode = "append")
-.format("jdbc")
-.option("driver", "com.mysql.jdbc.Driver")
- // Replace the host and port with that of your own and be sure to use the rewrite batch
-.option("url", "jdbc:mysql://127.0.0.1:4000/test?rewriteBatchedStatements=true")
-.option("useSSL", "false")
-// As tested, 150 is good practice
-.option(JDBCOptions.JDBC_BATCH_INSERT_SIZE, 150)
-.option("dbtable", s"cust_test_select") // database name and table name here
-.option("isolationLevel", "NONE") // recommended to set isolationLevel to NONE if you have a large DF to load.
-.option("user", "root") // TiDB user here
-.save()
-```
+Currently, TiSpark partially apply partition pruning on range partition.
 
-It is recommended to set `isolationLevel` to `NONE` to avoid large single transactions which might potentially lead to TiDB OOM.
+The partition pruning is applied when the partition expression of the range partition is one of the following:
 
-> **Note:**
->
-> When you use JDBC, the default value of `isolationLevel` is `READ_UNCOMMITTED`, which causes the error of unsupported isolation level transactions. It is recommended to set the value of `isolationLevel` to `NONE`.
++ column expression
++ `YEAR($argument)` where the argument is a column and its type is datetime or string literal
+  that can be parsed as datetime.
 
-## Statistics information
+If partition pruning is not applied, TiSpark's reading is equivalent to doing a table scan over all partitions.
+
+**Write into partition table**
+
+Currently, TiSpark only supports writing into the range and hash partition table under the following conditions:
 
-TiSpark uses TiDB statistic information for the following items:
++ the partition expression is column expression
++ the partition expression is `YEAR($argument)` where the argument is a column and its type is datetime or string literal
+  that can be parsed as datetime.
 
-1. Determining which index to ues in your query plan with the estimated lowest cost.
-2. Small table broadcasting, which enables efficient broadcast join.
+There are two ways to write into partition table:
 
-If you would like TiSpark to use statistic information, first you need to make sure that concerning tables have already been analyzed. Read more about [how to analyze tables](/statistics.md).
+1. Use datasource API to write into partition table which supports replace and append semantics.
+2. Use delete statement with Spark SQL.
 
-Starting from TiSpark 2.0, statistics information is default to auto load.
+> [!NOTE]
+> Currently the charset only supported is utf8mb4 and [`new_collations_enabled_on_first_bootstrap`](https://docs.pingcap.com/tidb/dev/tidb-configuration-file#new_collations_enabled_on_first_bootstrap)
+> need to be set to `false` in TiDB.
```

Comment:
not right for 3.1

Reply:
Partition table write still has this limitation